custom_components/modernforms/__init__.py

```diff
@@ -47,44 +53,33 @@ async def async_setup_entry(hass, config_entry):
 
   return True
 
-class ModernFormsBaseEntity(Entity):
-  def __init__(self, hass, device):
-    self.hass = hass
+class ModernFormsBaseEntity(CoordinatorEntity):
+  def __init__(self, coordinator, device):
+    super().__init__(coordinator)
     self.device = device
-    self.device._attach(self)
-
-    def update_action(time):
-      device.update_status()
-
-    async_call_later(hass, 0, update_action)
-    self.poll = async_track_time_interval(hass, update_action, device.interval)
 
-  def _device_updated(self):
-    self.schedule_update_ha_state()
+  async def perform_action_and_refresh(
+      self, action: MethodType, *args: Any, **kwargs: Any
+      ) -> bool:
+    await action(*args, **kwargs)
+    self.coordinator.async_set_updated_data(self.device.data)
 
-  @property
-  def should_poll(self):
-    return False
 
   @property
   def device_state_attributes(self):
     return self.device.data
 
 class ModernFormsDevice:
-  def __init__(self, name, host, has_light=False, interval=CONF_SCAN_INTERVAL):
+  def __init__(self, name, host, interval=CONF_SCAN_INTERVAL):
```

Comment:
Not all fans have lights. Why are you dropping this?

Reply:
has_light was unused outside `async_setup_entry` which should still skip the light entity if set to false